internal/service/budgets/budget_data_source.go

```diff
@@ -0,0 +1,345 @@
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"last_auto_adjust_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"budget_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"budget_limit": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"amount": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"unit": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"calculated_spend": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"actual_spend": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"amount": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"unit": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"cost_filter": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"values": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+			"cost_types": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"include_credit": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"include_discount": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"include_other_subscription": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"include_recurring": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"include_refund": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"include_subscription": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"include_support": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"include_tax": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"include_upfront": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"use_amortized": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"use_blended": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
```

Comment:
I think `name` should be required to align with the `GetBudget` API.

```suggestion
				Required: true,
```

Reply:
`name_prefix` may also need to be optional to completely align with how the resource creates the budget name that gets set in the `id` attribute. This same line can then be used in the data source's Read method.

https://github.com/hashicorp/terraform-provider-aws/blob/baf5cac858ed3379dcb3ffa6aa8a2be2d22fdfab/internal/service/budgets/budget.go#L297